crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -233,15 +231,52 @@ impl<F: Send + Clone> PostUpdateTracker<F, PaymentData<F>, types::PaymentsAuthor
             // The mandate is created on connector's end.
             let tokenization::SavePaymentMethodDataResponse {
                 payment_method_id,
-                mandate_reference_id,
+                connector_mandate_reference_id,
                 ..
             } = save_payment_call_future.await?;
-
+            payment_data.payment_method_info = if let Some(payment_method_id) = &payment_method_id {
+                match state
```

Comment:
we are creating the payment method in the previous function and then we are fetching it from the database?

Reply:
Cause in save_payment_method, we do a early return in some case, if we get a payment_id